### DISCUSSION TITLE
How to use dialog in class?

### DISCUSSION THREAD

**Discussion by HaoTian22** [flet-dev/flet#73] [Q&A] [answered]
I want to show a AlertDialog in a class function, but the dialog wasn't popped up after I click the button.
So, how to use it in class function?

By the way, I'm still a python beginner, so maybe my question is easy to you. But it does puzzle me for hours.
Here is my code.
```
import flet
from flet import AlertDialog, ElevatedButton, Page, Text, TextButton, UserControl

class App(UserControl):
	def build(self):
		return ElevatedButton("Open dialog", on_click=self.play)
	def play(self,e):
		dia = AlertDialog(
		title=Text('text'),
		actions=[TextButton("OK")],
		actions_alignment="end",
		)   
		self.dialog = dia
		dia.open = True
		self.update()

def main(page: Page):
	page.title = "AlertDialog examples"
	app=App()
	page.add(app)

flet.app(target=main,port=80)
```

**Comment by HaoTian22** [comment]
OK, I understand it.
Use self.controls.append(dia) in the class.

**Comment by FeodorFitsner** [comment, marked as answer]
Yep, you got it right. Another way would be:

```python
	def play(self,e):
		dia = AlertDialog(
		title=Text('text'),
		actions=[TextButton("OK")],
		actions_alignment="end",
		)   
		self.page.dialog = dia
		dia.open = True
		self.page.update()
```

**Reply by HaoTian22** [reply]
Thanks

**Comment by xiewencan** [comment]
Hello, I want to know how to access to the AlertDialog when you use `self.controls.append(dia)`

**Reply by ndonkoHenri** [reply]
You could loop through the controls until `isinstance(control, ft.AlertDialog)`.
